docs/SYNC_MODE.md

```diff
@@ -32,11 +32,68 @@ Taps can support two replication methods, and should decide which to use by chec
 
-| Metadata Keyword | Description  | 
+| Metadata Keyword | Description  |
 | ----------------- | ------- |
 | `inclusion` | Only applies to fields.  If this is set to `automatic`, the field should be replicated.  If this is set to `unsupported`, the field should not be replicated.  Can be written by a tap during discovery |
 | `selected` | If this is set to `True`, the stream (empty breadcrumb), or field should be replicated.  If `False`, the stream or field should be omitted.  This metadata is written by services outside the tap. |
+| `available` | The field may or may not be `selected`. |
 
+Streams shouldn’t be selected by default. We want them to have an `inclusion: available` property. Each field needs to have its own metadata object that labels the field as `inclusion: automatic` or `inclusion: available`.
+
+## Suggested Pattern for implementing stream selection
+You can get all the proper inclusion metadata during discovery by calling `get_standard_metadata()` as is done here:
+```
+ mdata = metadata.get_standard_metadata(
+            schema=<schema>,
+            key_properties=<key_properties>,
+            valid_replication_keys=<replication_keys>,
+            replication_method=<replication_method>,
+        )
+```
+The `mdata` object can then be passed to the catalog object.
+
+And then in the `sync` function, we'll want to iterate over only selected streams.
+We can get a list of the selected stream objects by calling `get_selected_streams()` on a singer-python `Catalog` object, as is done here:
+ ```
+selected_streams = catalog.get_selected_streams(state)
+
+for stream in selected_streams:
+    <sync stream>
+ ```
+
+## Suggested Pattern for implementing field selection
+You can get all the proper inclusion metadata during discovery by calling `get_standard_metadata()` as is done here:
+```
+ mdata = metadata.get_standard_metadata(
+            schema=<schema>,
+            key_properties=<key_properties>,
+            valid_replication_keys=<replication_keys>,
+            replication_method=<replication_method>,
+        )
+```
+The `mdata` object can then be passed to the catalog object.
+
+And then in the `sync` function, we'lll want to pass every record through the transformer, as is done here:
+ ```
+with Transformer() as transformer:
+    for rec in stream_object.sync():
+        singer.write_record(
+            <tap stream id>,
+            transformer.transform(
+                <record>, <schema dict>, <metadata dict>,
+            )
+        )
+```
+
+
+## How to handle child streams
+If there are child streams, this means that they rely on some piece of information from a corresponding parent.
+if child stream `CHILD` relies on parent stream `PARENT`'s id, then if we sync `CHILD` without `PARENT` being selected, the python process will error out with an unclear error message.
```

Comment:
I think this could be removed, there's no reason to mention the bad pattern first, since that'll stick in a reader's head, and we don't want that.